docker/bacalhau-dind/Dockerfile

```
@@ -2,9 +2,7 @@ FROM docker:dind
 
 # Install necessary packages
 RUN apk update && apk add --no-cache \
```

Hadolint (2.12.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)